### DISCUSSION TITLE
Insert Snippet is misbehaving if the snippet contains $

### DISCUSSION THREAD

**Discussion by amitjoshi438** [microsoft/vscode-discussions#644] [Extension Development QnA] [answered]
I am using the below function to insert code snippet from webview to editor:
`vscode.window.activeTextEditor?.insertSnippet(
                        new vscode.SnippetString(`${data.value}`)
                    );`
                    
  But for the below snippet, insert snippet omitted some parts:
  `GET [Organization URI]/api/data/v9.2/loanapplications?$select=applicantname,loanamount,creditscore&$orderby=creditscore desc&$top=3`                 
  
 Is this due to $ character in code snippet? How to handle this?
 
 Thanks!

**Comment by gjsjohnmurray** [comment, marked as answer]
See https://code.visualstudio.com/docs/editor/userdefinedsnippets#_grammar

I think you need to escape the $ characters in the text you are inserting, by preceding them with a \ character.
